x-pack/plugins/canvas/tasks/helpers/webpack.plugins.js

```diff
@@ -94,6 +94,16 @@ export function getWebpackConfig({ devtool, watch } = {}) {
           ignore: '**/__tests__/**',
         },
       ]),
+      function canvasStatsGenerator() {
+        if (!process.env.CANVAS_GENERATE_STATS) {
+          return;
+        }
+
+        this.hooks.done.tap('canvas_stats', stats => {
+          const content = JSON.stringify(stats.toJson());
+          writeFileSync('canvas_stats.json', content);
```

Comment:
Since there is a specific .gitignore for `/x-pack/canvas_stats.json` we should probably use an absolute path here. `resolve(__dirname, '../../../../canvas_stats.json')` but I also think it might make sense to write to the plugins/canvas directory.

Reply:
Good call... will address.